blkid/internal/filesystems/zfs/zfs.go

```diff
@@ -0,0 +1,101 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package zfs probes ZFS filesystems.
+package zfs
+
+//go:generate go run ../../cstruct/cstruct.go -pkg zfs -struct ZFSUB -input zfs.h -endianness LittleEndian
+
+import (
+	"fmt"
+
+	"github.com/siderolabs/go-blockdevice/v2/blkid/internal/magic"
+	"github.com/siderolabs/go-blockdevice/v2/blkid/internal/probe"
+)
+
+// https://github.com/util-linux/util-linux/blob/c0207d354ee47fb56acfa64b03b5b559bb301280/libblkid/src/superblocks/zfs.c
+const (
+	zfsUberblockCount = 128
+	zfsUberblockSize  = 1024
+	zfsLabelSize      = 1024
+	zfsVdevLabelSize  = 1024 * 256
+	zfsStartOffset    = 1024 * 128
+	zfsMinUberblocks  = 4 // Number of uberblocks to be found
```

Comment:
```suggestion
	zfsMinUberblocks  = 4 // Number of superblocks to be found
```

unless `uberblocks` is a thing :sweat_smile:

Reply:
Apparently sources other than util-linux also name it uberblock. We could name it superblock and comment it's uberblock in ZFS terms:
- https://github.com/util-linux/util-linux/blob/c0207d354ee47fb56acfa64b03b5b559bb301280/libblkid/src/superblocks/zfs.c
- https://people.freebsd.org/~gibbs/zfs_doxygenation/html/d9/d9f/structuberblock.html